filebeat/input/journald/pkg/journalfield/matcher.go

```diff
@@ -118,3 +136,86 @@ func ApplyMatchersOr(j journal, matchers []Matcher) error {
 
 	return nil
 }
+
+// ApplySyslogIdentifierMatcher adds unit based filtering to the journal reader.
+func ApplyUnitMatchers(j journal, units []string, kernel bool) error {
+	for _, unit := range units {
+		matchers := [][]Matcher{
+			[]Matcher{
+				MustBuildMatcher("systemd.unit=" + unit),
+			},
+			[]Matcher{
+				MustBuildMatcher("message_id=fc2e22bc6ee647b6b90729ab34a250b1"),
+				MustBuildMatcher("journald.uid=1"),
+				MustBuildMatcher("journald.coredump.unit=" + unit),
+			},
+			[]Matcher{
+				MustBuildMatcher("journald.pid=1"),
+				MustBuildMatcher("journald.unit=" + unit),
```

Comment:
Seems like this input can panic if misconfigured. Like any unit value that contains a `=` will crash (e.g. `units: ['unit=foo']` or just `units: ['=']`). I think it makes sense to use the `MustBuildMatcher` only where user controlled values are not involved.